Support_Your_Locals/Controllers/HomeController.cs

```diff
@@ -17,14 +18,12 @@ public HomeController(ILogger<HomeController> logger)
 
         public IActionResult Index()
         {
-            List<string> skelbimai = new List<string>();
-            skelbimai.Add("Kepejas");
-            skelbimai.Add("Auto-mechanikas");
-            skelbimai.Add("Auklyte");
-            skelbimai.Add("Grindu ploveja");
-            //ViewData["skelbimai"] = skelbimai;
+            var posts = new List<PostModel>()

```

Comment:
BusinessModel ne posts model su backendo pamatais dirbu aš. Laikykis draw.io diagramos schemos.

Reply:
Reikės pakeisti modelio struktūrą, nes atskirto table laikyti timesheetui visiškai neapsimoka. Efektyvesnis sprendimas yra JSON.

Reply:
Reikia ar nereikia pamatysim vėliau. Dabar daryk pagal diagramą (struktūrinių pakeitimų nederėtų daryti be daugumos komandoje žinios), jei prireiks pakeisim ateityje. Čia gaunasi over-engeenering pagalvok pats, profesionalai nevaro iškart in-depth, pirma padaro infrastruktūrą, paskui varo į sudėtingesnį kodavimą. Tu dar nepadarei index page, o jau varai į business post page. Dėl to gaunasi milijonas hardcoding atvejų ir keisti ką nors bus vargas. Laikykis susitarimo, pasakei koduosi index page, tai ir daryk, kas sutarta. Prieš eidamas gilyn, ištaisyk hardcoding ir šiam ir praeitam pull request, kurį sumerginai neištaisęs.

Reply:
Evaldai, čia ne joks over-engineerinimas, over-engineerimas yra daryti table dedikuota vien working valandom, va čia tai yra čystas overkillas. Aps-for kaip sakiau čia nereikia, nes PostsControlleris yra naudojamas kaip web servisas, į kurį kreipiamasi per JavaScript.

Esmė yra tokia - kaip tu vien su ASP.NET nusiųsi į backendą TimeSheetą? Kaip tu jam nurodysi siųsti tik tas dienas, kurios yra pažymėtos checkboxu? Šitos problemos solution yra AJAX. 

Taip yra žymiau lengviau nusiųsti šitą formą į kontrollerį, nes su JavaScriptu nurodau tiksliai kokiu formatu ir kokius duom reikia nusiųst. O dėl index page, wtf? Kaip padarysim index page jeigu neturim sukurtų businessų? Dėstytojui buvom pasakė, kad padarysim kad būtų toks workflowas: sukuri postą, jį displayina index page, čia yra viskas logical and natural.

Reply:
Kaip jau rašiau Teams, Business model fieldas "timesheet" turės būti string tipo, nes jame bus talpinamas JSON formato timesheetas, kuriame bus nurodytos visos dienos, kuriomis dirbama bei valandos/minutės nuo iki. Labai easy pafetchinti ir nuparsinti su JSON.parse funkcija

Reply:
Gali daryti su JSON ir AJAX, bet tegu darbo valandos būna užregistruotos ir išlieka kaip atskira duomenų lentelė duombazėje.

Reply:
Ir koks to tikslas? Kodėl tu nori vieną string recordą laikyti kaip atskirą lentelę?